pkg/log/zap/logger.go

```diff
@@ -75,6 +75,13 @@ func getConfig(destWriter io.Writer) config {
 	}
 	if levelVal.set {
 		c.level = zap.NewAtomicLevelAt(levelVal.level)
+
+		// Disable sampling when we are in debug mode. Otherwise, this will
+		// cause index out of bounds errors in the sampling code.
```

Comment:
Is this the fix for the bug mentioned [here](https://github.com/operator-framework/operator-sdk/pull/873#discussion_r252335389)?

Reply:
Yeah, I bet that this the case.

Reply:
I'm pretty sure it is. Nice find @jmrodri!

With this being a possible source of runtime panics, can we add some unit tests to make sure we don't introduce a regression later?